OTC-232: Integrate DB Stored Procedures

sniedzielski opened the pull request:
TICKET: https://openimis.atlassian.net/browse/OTC-232

In that PR I want to integrate Stored Procedures both in "imis_sql_online.sql" (only CREATE PROCEDURE) and "latest migrations" (CREATE PROCEDURE with DROP ... IF EXISTS).

So as to make read this PR easier I added commented strings above modified SP - "-- OTC-232: Integrate DB Stored Procedures" and missing ones - "-- OTC-232: Integrate DB Stored Procedures: missing part". (both in "latest migration and imis_online" files)

Short summary:
a) added SP (missing ones) both in "latest migration and imis_online":
- [dbo].[uspAPIGetClaimOverview]
- [dbo].[uspGetClaimStats]
- [dbo].[uspSentRequestGetControlNumber]
- [dbo].[uspSSRSGetClaimHistoryClaimsAllDetails]
- [dbo].[uspSSRSGetClaimHistoryClaimsAndRejectionDetails]
- [dbo].[uspSSRSGetClaimHistoryClaimsOnly]
- [dbo].[uspSSRSGetClaimHistoryReport]
- [dbo].[uspSSRSGetClaimOverviewClaimsAllDetails]
- [dbo].[uspSSRSGetClaimOverViewClaimsOnly]
- [dbo].[uspSSRSGetClaimOverViewRejectedServiceItem]

b) modified SP (existing ones which are different etc) both in "latest migration and imis_online":
- [dbo].[uspPolicyValue]
- [dbo].[uspAcknowledgeControlNumberRequest]
- [dbo].[uspAddInsureePolicyOffline]
- [dbo].[uspAPIEnterContribution]
- uspAPIGetCoverage
- [dbo].[uspAPIRenewPolicy]
- [dbo].[uspBackupDatabase]
- uspRelativeIndexCalculationMonthly - added old version "uspRelativeIndexCalculationMonthlyOld"
- [dbo].[uspConsumeEnrollments]
- [dbo].[uspExportOffLineExtract6]
- uspFeedbackPromptSMS
- [dbo].[uspGetPolicyRenewals] 
- [dbo].[uspImportOffLineExtract4]
- [dbo].[uspIsValidRenewal]
- lastDateForPayment
- [uspMatchPayment]
- [dbo].[uspPhoneExtract]
- [dbo].[uspPolicyInquiry] 
- [dbo].[uspPolicyRenewalInserts
- uspProcessSingleClaimStep1
- uspReceiveControlNumber
- [dbo].[uspRefreshAdmin]
- [dbo].[uspServiceItemEnquiry]
- [dbo].[uspSSRSCapitationPayment]
- [uspSSRSEnroledFamilies]
- [dbo].[uspSSRSGetClaimOverview] (and also [dbo].[uspSSRSGetClaimOverView] vs [dbo].[uspSSRSGetClaimOverview])
- [dbo].[uspSSRSGetMatchingFunds]
- uspSSRSPaymentCategoryOverview
- [dbo].[uspSubmitClaims]
- [dbo].[uspUpdateClaimFromPhone]

The rest ones are without changes.

dragos-dobre commented on Migration script/openIMIS migration latest.sql:
```diff
@@ -5849,3 +5985,6030 @@ NEXT_POLICY:
+
+-- 4/11/2019
+
+CREATE PROCEDURE [dbo].[uspAPIGetClaimOverview]
```
I don't find this SP called in the REST API or WebApp. Can you check if this is actual used somewhere?

sniedzielski replied:
Ok. I will check this and confirm this whether is somewhere in that apps

sniedzielski commented on Migration script/openIMIS migration latest.sql:
```diff
@@ -3624,277 +3624,413 @@ CREATE PROCEDURE [dbo].[uspSSRSStatusRegister]
+SET QUOTED_IDENTIFIER ON
+GO
 
+CREATE PROCEDURE [dbo].[uspSSRSGetClaimOverView]
```
Ok.

sniedzielski commented on Migration script/openIMIS migration latest.sql:
```diff
@@ -5849,3 +5985,6030 @@ NEXT_POLICY:
+GO
+
+--29/10/2019
+CREATE PROCEDURE [dbo].[uspSSRSGetClaimOverviewClaimsAllDetails]
```
Ok. I will remove this!

sniedzielski commented on Migration script/openIMIS migration latest.sql:
```diff
@@ -5849,3 +5985,6030 @@ NEXT_POLICY:
+SET QUOTED_IDENTIFIER ON
+GO
+
+CREATE PROCEDURE [dbo].[uspSSRSGetClaimOverViewClaimsOnly]
```
Ok. I will remove this!

sniedzielski commented on Migration script/openIMIS migration latest.sql:
```diff
@@ -5849,3 +5985,6030 @@ NEXT_POLICY:
+GO
+
+
+CREATE PROCEDURE [dbo].[uspSSRSGetClaimOverViewRejectedServiceItem]
```
Ok. I will remove this!

sniedzielski commented:
@dragos-dobre I've updated PR. Here below what I changed:
a) removed [dbo].[uspSSRSGetClaimHistoryClaimsAllDetails], [dbo].[uspSSRSGetClaimHistoryClaimsAndRejectionDetails] and [dbo].[uspSSRSGetClaimHistoryClaimsOnly] - because they are a part of [uspSSRSGetClaimHistoryReport]
b) removed  [dbo].[uspSentRequestGetControlNumber] - no sign both in web app and c# rest api
c) removed [dbo].[uspGetClaimStats] - no sign both in web app and c# rest api
d) removed [dbo].[uspAPIGetClaimOverview] -  no sign both in web app and c# rest api
e) removed [dbo].[uspSSRSGetClaimOverviewClaimsAllDetails], [dbo].[uspSSRSGetClaimOverViewClaimsOnly], [dbo].[uspSSRSGetClaimOverViewRejectedServiceItem], [dbo].[uspSSRSGetClaimOverView] - because they are a part of [dbo].[uspSSRSGetClaimOverview]

sniedzielski commented:
@dragos-dobre I've just updated my PR by adding @dborowiecki changes from https://github.com/openimis/database_ms_sqlserver/pull/62. 
included:
- [dbo].[uspCreateCapitationPaymentReportData]
- [dbo].[uspSSRSRetrieveCapitationPaymentReportData]
Therefore we can continue compare SQL Storage Procedures between TNZ and MV.

delcroip commented:
@dragos-dobre do we need to discuss on this one ?

dragos-dobre commented:
@delcroip Not really. I need to check which SP from the list is used in the REST API. We also have to test the ePayment with empty DB (tested with CHF db).

dragos-dobre commented:
No missing SP were reported so we close this PR.
